errors.go

```diff
@@ -7,6 +7,43 @@ import (
 	"github.com/pkg/errors"
 )
 
+type JobNotFoundError struct {
+	msg string
+}
+
+func (e *JobNotFoundError) Error() string { return e.msg }
+
+// NewJobNotFoundError creates a new error indicating that a job could not be
+// found in the queue.
+func NewJobNotFoundError(msg string) error { return &JobNotFoundError{msg: msg} }
```

Comment:
Should this return `*JobNotFoundError`?

Reply:
It already does.

Reply:
But the return type is `error`. Aren't we doing the accept interfaces return structs thing?

Reply:
Oh I see you mean the function signature. In this case, I don't think it matters because the only meaningful way you can use it is as an error. But it's not really an issue to return the struct as-is, so I changed the return value.